pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -395,6 +395,10 @@ func (r *repository) getImageOfImageStream(dgst digest.Digest) (*imageapi.Image,
 		return nil, nil, wrapKStatusErrorOnGetImage(r.name, dgst, err)
 	}
 
+	// We are going to pretend that we've got the image from the same source
+	// as the client.
+	image.DockerImageReference = tagEvent.DockerImageReference
```

Comment:
I don't think we can afford this. Occasionally, registry updates the image (e.g. when it removes its manifest). We don't want to change the original `dockerImageReference` as well.

I don't advocate against using `tagEvent.DockerImageReference`. It's correct for most of the cases, but I don't think we can set it on the image object itself.